server/service/osquery.go

```diff
@@ -707,7 +707,9 @@ func (svc *Service) detailQueriesForHost(ctx context.Context, host *fleet.Host)
 			queryName := hostDetailQueryPrefix + name
 			queries[queryName] = query.Query
 			if query.QueryFunc != nil && query.Query == "" {
-				queries[queryName] = query.QueryFunc(ctx, svc.logger, host, svc.ds)
+				if query, ok := query.QueryFunc(ctx, svc.logger, host, svc.ds); ok {
+					queries[queryName] = query
+				}
```

Comment:
If `!ok` then you need to `continue` to not add the discovery query to `discovery`.

Reply:
Done in 817a864a4865d861baa0ae9bbb15d45a765f6714